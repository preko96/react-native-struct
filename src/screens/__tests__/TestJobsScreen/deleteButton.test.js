import React from 'react'
import DeleteButton from '../../TestJobsScreen/deleteButton'

let wrapper

const mockProps = {
	secId: '0',
	rowId: '0',
	rowMap: {
		'00': {
			props: {
				closeRow: jest.fn()
			}
		}
	},
	onPress: jest.fn(),
}

beforeEach(()=>{
	jest.clearAllMocks()
	wrapper = shallow(<DeleteButton {...mockProps} />)
})

it('renders without crashing', () => {
	expect(wrapper).toMatchSnapshot()
})

it('close side and call the onPress callback', () => {
	wrapper.simulate('Press')
	expect(mockProps.rowMap['00'].props.closeRow).toHaveBeenCalledTimes(1)
	expect(mockProps.onPress).toHaveBeenCalledTimes(1)
}) 
