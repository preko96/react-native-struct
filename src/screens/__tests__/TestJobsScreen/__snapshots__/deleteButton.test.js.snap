// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(Button)
  danger={true}
  full={true}
  onPress={[Function]}
>
  <Styled(Icon)
    active={true}
    name="trash"
  />
</Styled(Button)>
`;
